File: pkg/java/src/main/java/dev/openfga/language/errors/UnsupportedDSLNestingException.java
package dev.openfga.language.errors;

public class UnsupportedDSLNestingException extends RuntimeException {
    public UnsupportedDSLNestingException(String typeName, String relationName) {
        super(String.format(
                "the '%s' relation definition under the '%s' type is not supported by the OpenFGA DSL syntax yet",
                relationName, typeName));
    }
}

File: pkg/csharp/src/OpenFga.Language/Errors/UnsupportedDSLNestingException.cs
namespace OpenFga.Language.Errors;

/// <summary>
/// Exception thrown when attempting to transform a JSON authorization model to DSL format
/// that contains unsupported nesting patterns in relation definitions.
/// </summary>
/// <param name="typeName">The name of the type containing the unsupported relation</param>
/// <param name="relationName">The name of the relation with unsupported nesting</param>
public class UnsupportedDslNestingException(string typeName, string relationName)
    : Exception($"the '{relationName}' relation definition under the '{typeName}' type is not supported by the OpenFGA DSL syntax yet");